File: Homework2_Derezovskiy/Homework2_Derezovskiy/ball.cpp
//------------------------------------------------------------------------------
// ball_In.cpp - содержит процедуру ввода параметров
// для уже созданного шара
//------------------------------------------------------------------------------

#define _USE_MATH_DEFINES
#include "math.h"
#include "ball.h"

//------------------------------------------------------------------------------
// Ввод параметров шара из файла
void Ball::In(std::istream& ifst) {
    ifst >> r >> materialDensity;
}

// Случайный ввод параметров шара
void Ball::InRnd() {
    r = Shape::rnd20.Get();
    materialDensity = Shape::rnd20.Get();
}

//------------------------------------------------------------------------------
// Вывод параметров шара в форматируемый поток
void Ball::Out(std::ostream& ofst) {
    ofst << "It is Ball: r = " << r << ". Material density = " << materialDensity
        << ". Square = " << Square() << "\n";
}

//------------------------------------------------------------------------------
// Вычисление площади поверхности шара
double Ball::Square() {
    return 4.0 * M_PI * (r * r);
}

File: Homework3_Derezovskiy/src/ball.py
import random
import math

class Ball:
    def __init__(self):
        self.r = 0
        self.materialDensity = 0

    def In(self, strArray):
        self.r = int(strArray[1])
        self.materialDensity = int(strArray[2])

    def InRnd(self):
        self.r = random.randint(1, 20)
        self.materialDensity = random.randint(1, 20)

    def Write(self, ostream):
        ostream.write("Ball: r = {}, materialDensity = {}, Square = {}".
                      format(self.r, self.materialDensity, self.Square()))
        pass

    def Square(self):
        return 4.0 * math.pi * (self.r * self.r);
        pass
